.github/workflows/test.yaml

```diff
@@ -79,6 +79,53 @@ jobs:
         with:
           pipeline_version: ${{ matrix.pipeline-version }}
 
+      # setting up Knative Serving
+      - name: setup-serverless
+        run: |
+          curl -O -L "https://github.com/sigstore/cosign/releases/latest/download/cosign-linux-amd64"
+          sudo mv cosign-linux-amd64 /usr/local/bin/cosign
+          sudo chmod +x /usr/local/bin/cosign
+          sudo apt-get install jq
+          curl -sSL https://github.com/knative/serving/releases/download/knative-v1.13.1/serving-core.yaml \
+            | grep 'gcr.io/' | awk '{print $2}' | sort | uniq \
+            | xargs -n 1 \
+              cosign verify -o text \
+                --certificate-identity=signer@knative-releases.iam.gserviceaccount.com \
+                --certificate-oidc-issuer=https://accounts.google.com
+          kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.13.1/serving-crds.yaml
+          kubectl apply -f https://github.com/knative/serving/releases/download/knative-v1.13.1/serving-core.yaml
+          kubectl apply -f https://github.com/knative/net-kourier/releases/download/knative-v1.13.0/kourier.yaml
+          kubectl patch configmap/config-network \
+            --namespace knative-serving \
+            --type merge \
+            --patch '{"data":{"ingress-class":"kourier.ingress.networking.knative.dev"}}'          
+          sleep 60
+          cat <<EOF | kubectl apply -f -
+          apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Can you also add a comment on why this roles are required ???

Reply:
yep added comment, roles were needed here to create service in default namespace.